In [20]:
import seaborn as sns
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
from google.colab import drive

drive.mount("/content/drive")

Mounted at /content/drive


## Подгрузим данные

In [19]:
recipe_data = pd.read_csv(
    filepath_or_buffer="/content/drive/MyDrive/Colab Notebooks/Hackathon/mock-med-data/data_recipe.csv"
)
recipe_data.head(4)

,id_пациента,дата_рецепта,код_диагноза,код_препарата,id_пациента.1
0,24419779,2022-03-31 11:52:30,F06,200900018,1226228
1,17890856,2019-06-04 18:41:10,I11.9,201300167,62857
2,18749669,2019-10-29 12:35:31,K86.1,201300027,1186134
3,31412579,2024-07-30 17:06:59,I67.8,202200024,1927759


## Наличие Nan

In [5]:
print(recipe_data.isna().sum())
print("\n")

print(f"{(recipe_data.isna().sum() / len(recipe_data)) * 100} %(percentage of nan's)", "\n")

id_пациента         0
дата_рецепта     1178
код_диагноза        0
код_препарата       0
id_пациента.1       0
dtype: int64


id_пациента      0.0000
дата_рецепта     0.1178
код_диагноза     0.0000
код_препарата    0.0000
id_пациента.1    0.0000
dtype: float64 %(percentage of nan's) 



### Информация по типам

In [6]:
recipe_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000000 entries, 0 to 999999
Data columns (total 5 columns):
 #   Column         Non-Null Count    Dtype 
---  ------         --------------    ----- 
 0   id_пациента    1000000 non-null  int64 
 1   дата_рецепта   998822 non-null   object
 2   код_диагноза   1000000 non-null  object
 3   код_препарата  1000000 non-null  int64 
 4   id_пациента.1  1000000 non-null  int64 
dtypes: int64(3), object(2)
memory usage: 38.1+ MB


In [10]:
recipe_data.describe()

,id_пациента,код_препарата,id_пациента.1
count,1.000000e+06,1.000000e+06,1.000000e+06
mean,2.522643e+07,2.015874e+08,1.012471e+06
std,4.443059e+06,4.300166e+05,5.833790e+05
min,1.769453e+07,2.000522e+08,2.800000e+01
25%,2.139583e+07,2.012002e+08,5.166138e+05
50%,2.505255e+07,2.016001e+08,1.012891e+06
75%,2.901513e+07,2.020000e+08,1.521807e+06
max,3.320412e+07,2.025000e+08,2.023358e+06


## Форма датасета

In [7]:
recipe_data.shape

(1000000, 5)

## Количество уникальных лэйблов

In [8]:
for col in recipe_data.columns:
    print(f"{col} : ", {len(pd.unique(recipe_data[col])): pd.unique(recipe_data[col])}, "\n")

id_пациента :  {847670: array([24419779, 17890856, 18749669, ..., 32582840, 20849486, 19498691])} 

дата_рецепта :  {839286: array(['2022-03-31 11:52:30', '2019-06-04 18:41:10',
       '2019-10-29 12:35:31', ..., '2024-11-28 15:48:54',
       '2020-10-08 11:59:55', '2020-02-17 16:04:11'], dtype=object)} 

код_диагноза :  {2433: array(['F06', 'I11.9', 'K86.1', ..., 'G81.0', 'C06.8', 'A18.5'],
      dtype=object)} 

код_препарата :  {2072: array([200900018, 201300167, 201300027, ..., 201200112, 202400067,
       201100056])} 

id_пациента.1 :  {114228: array([1226228,   62857, 1186134, ..., 1165650, 1299255, 1038071])} 



## Самые часто встречающиеся значения

In [9]:
for idx, col in enumerate(recipe_data.columns):
    print(recipe_data.iloc[:, idx].value_counts(ascending=False), "\n")

id_пациента
23928071    46
23631331    42
18789347    41
18849543    39
21416610    38
            ..
18386888     1
23804423     1
30397693     1
21749270     1
30642111     1
Name: count, Length: 847670, dtype: int64 

дата_рецепта
2022-01-26 09:59:06    46
2021-12-10 09:36:13    42
2019-11-05 18:15:23    41
2019-11-13 15:51:49    39
2021-01-14 09:48:50    38
                       ..
2019-05-23 10:56:37     1
2022-11-24 09:20:48     1
2020-07-08 20:49:18     1
2021-11-01 15:36:59     1
2023-06-02 10:22:58     1
Name: count, Length: 839285, dtype: int64 

код_диагноза
I11.9    141200
I25.1     70462
E11.7     59735
E10.9     57911
E11.9     54623
          ...  
I68.9         1
L30.0         1
T92.5         1
J98.4         1
I08.1         1
Name: count, Length: 2433, dtype: int64 

код_препарата
201700016    22373
201700002    20439
201000730    16676
201000698    14407
202000003    13318
             ...  
202400167        1
202400150        1
201200112        1
202400067        1
2

## Выводы:
  - 0.1178 процентов пропущенных в `дата_рецепта`, что достаточно мало и также можно его разделить на день месяц год
  - Остальные признаки крисстально чисты, но `id_пациента.1` - это реальный индекс по id , а `id_пациента` - исключить из датасета
  -  все отсальное более менее очевидно, что они связаны с остальными датасетами, можно дополнить эту таблицу информацией из соседних `код_диагноза`(diagnosis_data), `код_препарата`(medicine_data)
